cairo/tests/test_serde.py

```diff
@@ -173,3 +174,16 @@ def test_type(
         base = segments.gen_arg([gen_arg(type_, b)])
         result = serde.serialize(to_cairo_type(type_), base, shift=0)
         assert result == b
+
+    def test_exceptions(self, to_cairo_type, segments, serde, gen_arg):
+        # No Exception - cast(base, felt) == 0
+        base = gen_arg(EthereumException, None)
+        result = serde.serialize(to_cairo_type(EthereumException), base, shift=0)
+        assert result is None
+
+        # Test error case - cast(base, felt) != 0
+        expected_error = EthereumException("Some Error")
+        base = segments.gen_arg([gen_arg(EthereumException, expected_error)])
+        with pytest.raises(EthereumException) as exception:
+            serde.serialize(to_cairo_type(EthereumException), base, shift=0)
+        assert str(exception.value) == str(expected_error)
```

Comment:
plz do unit testing, no two tests in the same test

Reply:
can't we have hypothesis to generate errors ? I'd have liked to have the same as `test_type` with a list of Exceptions as we add some more